x-pack/platform/plugins/private/cross_cluster_replication/public/__jest__/client_integration/helpers/auto_follow_pattern_list.helpers.js

```diff
@@ -24,93 +26,56 @@ const testBedConfig = {
-    const indexLastColumn = rows[index].columns.length - 1;
-    const tableCellActions = rows[index].columns[indexLastColumn].reactWrapper;
-
-    let button;
-    if (action === 'delete') {
-      button = findTestSubject(tableCellActions, 'deleteButton');
-    } else if (action === 'edit') {
-      button = findTestSubject(tableCellActions, 'editButton');
-    }
-
-    if (!button) {
-      throw new Error(`Button for action "${action}" not found.`);
-    }
-
-    button.simulate('click');
-  };
-
-  const clickAutoFollowPatternAt = (index = 0) => {
-    const { rows } = testBed.table.getMetaData(EUI_TABLE);
-    const autoFollowPatternLink = findTestSubject(
-      rows[index].reactWrapper,
-      'autoFollowPatternLink'
-    );
-    autoFollowPatternLink.simulate('click');
-  };
-
-  const clickPaginationNextButton = () => {
-    testBed.find('autoFollowPatternListTable.pagination-button-next').simulate('click');
-  };
+      };
+    },
+    defaultProps: props,
+  });
 
   return {
-    ...testBed,
+    ...result,
+    // Helper actions for this specific page
     actions: {
-      selectAutoFollowPatternAt,
-      clickBulkDeleteButton,
-      clickConfirmModalDeleteAutoFollowPattern,
-      clickRowActionButtonAt,
-      clickAutoFollowPatternAt,
-      getPatternsActionMenuItemText,
-      clickPatternsActionMenuItem,
-      clickPaginationNextButton,
+      async selectAutoFollowPatternAt(index) {
+        const rows = getTableRows('autoFollowPatternListTable');
+        const checkbox = within(rows[index]).getByRole('checkbox');
+        await result.user.click(checkbox);
+      },
+
+      async clickBulkDeleteButton() {
+        const btn = screen.getByTestId('autoFollowPatternActionMenuButton');
+        await result.user.click(btn);
+
+        const contextMenu = screen.getByTestId('autoFollowPatternActionContextMenu');
+        const deleteBtn = within(contextMenu).getAllByRole('button')[1];
+        await result.user.click(deleteBtn);
+      },
+
+      async clickConfirmModalDeleteAutoFollowPattern() {
+        const confirmBtn = screen.getByTestId('confirmModalConfirmButton');
+        await result.user.click(confirmBtn);
+        // Wait for delete HTTP request and list reload
+        await act(async () => {
+          // eslint-disable-next-line no-undef
+          await jest.runOnlyPendingTimersAsync();
+        });
+      },
+
+      async clickAutoFollowPatternAt(index) {
+        const link = screen.getAllByTestId('autoFollowPatternLink')[index];
+        await result.user.click(link);
+      },
+
+      async clickPaginationNextButton() {
+        const table = screen.getByTestId('autoFollowPatternListTable');
+        const nextBtn = within(table).getByTestId('pagination-button-next');
+        await result.user.click(nextBtn);
+      },
+
+      async search(value) {
+        const input = screen.getByTestId('autoFollowPatternSearch');
+        await result.user.clear(input);
```

Comment:
nit: can be simplified here and in other places:

```suggestion
    await user.type(nameField, 'updated value', { selectAll: true });
```

Reply:
this api is not in the current lib unforuntately